ykcompile/src/lib.rs

```diff
@@ -94,77 +97,133 @@ impl CompiledTrace {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+enum Location {
+    Register(u8),
+    Stack(i32),
```

Comment:
For `Stack`, where is the offset relative to?

Reply:
It's relative to `rsp`. At the beginning of the trace we do `sub rsp`, so to get to a stack location we need to do `rsp + offset`.

Reply:
OK, there's one detail missing in my mental model: how do we make sure the offsets remain correct when more stuff gets spilled?

It looks like (and I may be wrong) that it'd work like this. Suppose:

 - registers are full
 - allocate for local var `a`. Is given `Location::stack(0)`.
 - allocate for local var `b`. Is given `Location::stack(8)`.

But that doesn't look right...

Reply:
If you mean making sure that the same variable always get allocated to the same offset, then that isn't implemented yet. This PR was already getting quite big, so I was going to do that in another one.

Reply:
That wasn't what I meant, but we resolved this comment in the team meeting I think.